azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilterAutoConfiguration.java

```diff
@@ -60,7 +64,12 @@ public AADAuthenticationFilterAutoConfiguration(AADAuthenticationProperties aadA
     @ConditionalOnExpression("${azure.activedirectory.session-stateless:false} == false")
     public AADAuthenticationFilter azureADJwtTokenFilter() {
         LOG.info("AzureADJwtTokenFilter Constructor.");
-        return new AADAuthenticationFilter(aadAuthProps, serviceEndpointsProps, getJWTResourceRetriever());
+        if (getJWKSetCache() != null) {
```

Comment:
This method will always return null. How can you get the specific JWKSetCache customized by yourself?

Reply:
I am also wondering about this. Trying to figure this part out

Reply:
I think with the below changes I suggested we might not need the null check. We would always add it as the 4th parameter. And we could default the lifespan to 5 minutes. That's what it gets defaulted to if you don't pass it in as per the documentation so that would be the same behavior unless you specify in the properties file

Reply:
Sounds good.